src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/controller/admin/AdminApiIntTest.kt

```diff
@@ -0,0 +1,120 @@
+package uk.gov.justice.digital.hmpps.personrecord.controller.admin
+
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.springframework.http.MediaType
+import uk.gov.justice.digital.hmpps.personrecord.api.model.admin.AdminReclusterRecord
+import uk.gov.justice.digital.hmpps.personrecord.config.WebTestBase
+import uk.gov.justice.digital.hmpps.personrecord.model.types.SourceSystemType
+import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType
+import java.util.UUID
+
+class AdminApiIntTest : WebTestBase() {
+
+  @Nested
+  inner class MissingRecord {
+
+    @Test
+    fun `should throw not found error when cluster not found in list`() {
+      val uuid = UUID.randomUUID()
+      val request = listOf(AdminReclusterRecord(SourceSystemType.COMMON_PLATFORM, uuid.toString()))
+
+      webTestClient.post()
+        .uri(ADMIN_RECLUSTER_URL)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isOk
+
+      checkTelemetry(
+        TelemetryEventType.CPR_ADMIN_RECLUSTER_TRIGGERED,
+        mapOf("UUID" to uuid.toString()),
+        times = 0,
+      )
+    }
+
+    @Test
+    fun `should not recluster records that have been merged`() {
+      val person = createPersonWithNewKey(createRandomProbationPersonDetails())
+      val mergedPerson = createPersonWithNewKey(createRandomProbationPersonDetails())
+
+      mergeRecord(mergedPerson, person)
+
+      mergedPerson.assertMergedTo(person)
+
+      val request = listOf(AdminReclusterRecord(SourceSystemType.COMMON_PLATFORM, mergedPerson.crn!!))
+
+      webTestClient.post()
+        .uri(ADMIN_RECLUSTER_URL)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isOk
+
+      checkTelemetry(
+        TelemetryEventType.CPR_ADMIN_RECLUSTER_TRIGGERED,
+        mapOf("UUID" to person.personKey?.toString()),
+        times = 0,
+      )
+    }
+  }
+
+  @Nested
+  inner class SuccessfulProcessing {
+
+    @BeforeEach
+    fun beforeEach() {
+      stubPersonMatchUpsert()
+      stubPersonMatchScores()
+    }
+
+    @Test
+    fun `should recluster single cluster`() {
```

Comment:
should recluster single record
```suggestion
    fun `should recluster single person record`() {
```